pkg/liqoctl/add/add_test.go

```diff
@@ -0,0 +1,127 @@
+package add
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fake2 "sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	discoveryv1alpha1 "github.com/liqotech/liqo/apis/discovery/v1alpha1"
+	"github.com/liqotech/liqo/pkg/consts"
+	foreigncluster "github.com/liqotech/liqo/pkg/utils/foreignCluster"
+)
+
+const (
+	localClusterID    string = "local-cluster-id"
+	remoteTokenPrefix string = "remote-token-"
+)
+
+var (
+	ctx       context.Context
+	clientset *fake.Clientset
+	k8sClient client.Client
+)
+
+func TestAddCommand(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Test Parameters Fetching")
+}
+
+var _ = Describe("Extract elements from apiServer", func() {
+	BeforeEach(func() {
+		clientset, k8sClient = setUpEnvironment(&ClusterArgs{
+			ClusterName:    "cluster-local",
+			ClusterToken:   "test-token",
+			ClusterAuthURL: "https://example.com",
+			ClusterID:      "test-cluster-id",
+			Namespace:      "liqo",
+		})
+	})
+
+	When("Add a cluster with a ClusterID equal to the local one", func() {
+		It("Add should fail", func() {
+			err := processAddCluster(ctx, &ClusterArgs{
+				ClusterName:    "cluster-local",
+				ClusterToken:   "test-token",
+				ClusterAuthURL: "https://example.com",
+				ClusterID:      localClusterID,
+				Namespace:      "liqo",
+			}, clientset, k8sClient)
+			Expect(err).To(HaveOccurred())
+			if err != nil {
+				Expect(err.Error()).To(BeEquivalentTo(sameClusterError))
+				return
+			}
+		})
+	})
+	When("Add a cluster with a ClusterID", func() {
+		It("Should create the following resources", func() {
+			t := &ClusterArgs{
+				ClusterName:    "cluster1bis",
+				ClusterToken:   "test-token",
+				ClusterAuthURL: "https://example.com",
+				ClusterID:      "test-cluster-id2",
+				Namespace:      "liqo",
+			}
+			Expect(processAddCluster(ctx, t, clientset, k8sClient)).To(Succeed())
+			AssertValidAddition(t)
+		})
+		When("Add a cluster with the same ClusterID", func() {
+			It("Should enforce the new values", func() {
+				r := &ClusterArgs{
+					ClusterName:    "cluster1bis",
+					ClusterToken:   "test-token",
+					ClusterAuthURL: "https://example.com",
+					ClusterID:      "test-cluster-id2",
+					Namespace:      "liqo",
+				}
+				Expect(processAddCluster(ctx, r, clientset, k8sClient)).To(Succeed())
+				AssertValidAddition(r)
+			})
+		})
+	})
+
+})
+
+func AssertValidAddition(t *ClusterArgs) {
```

Comment:
Nit. To avoid polluting the package (especially since this is not a dedicated package), the utility functions could be defined as variables inside the Describe block.